pkg/engine/globalcontext/store/store.go

```diff
@@ -0,0 +1,52 @@
+package store
+
+import (
+	"sync"
+)
+
+type Store interface {
+	Set(key string, val Entry) bool
+	Get(key string) (Entry, bool)
+	Delete(key string)
+}
+
+type store struct {
+	sync.RWMutex
+	store map[string]Entry
+}
+
+func New() Store {
+	return &store{
+		store: make(map[string]Entry),
+	}
+}
+
+func (l *store) Set(key string, val Entry) bool {
+	l.Lock()
+	defer l.Unlock()
+	old := l.store[key]
+	// If the key already exists, skip it before replacing it
+	if old != nil {
+		val.Stop()
+	}
+	l.store[key] = val
+	_, ok := l.store[key]
+	return ok
+}
```

Comment:
when is this going to return false ?

Reply:
never 😅 We can remove the return statement

Reply:
Fixed